In [1]:
import pandas as pd
import numpy as np

In [70]:
school_survey = pd.read_csv('data/36423-0001-Data.tsv',sep='\t')
student_survey = pd.read_csv('data/36423-0002-Data.tsv',sep='\t')

X2UNIV2A: X2 Base year status and how sample member entered F1 sample

Indicates the base year status (respondent, non-respondent, questionnaire ineligible) for base year sample members.

Base year respondent = 1

In [148]:
check1 = student_survey['X2UNIV2A'].value_counts()
check1

1    21444
2     2059
Name: X2UNIV2A, dtype: int64

In [152]:
filter1 = student_survey[student_survey['X2UNIV2A'] == 1]

Yay! From paper, we have first repeated result:

"HSLS:09 began with a nationally rep- resentative sample of ninth graders in fall 2009 (21,444 students from 944 schools) and followed them through postsecondary edu- cation."

Now we need to check for *first time* 9th graders, as per Appendix A

This is the relevant variable: S1GRD0809, "S1 B01 Grade 9th grader was in last year (2008-09)"

1 7th Grade 64 0.3 %

2 8th Grade 20178 85.9 %

3 9th Grade 1124 4.8 %

4 In an ungraded program 61 0.3 %

In [153]:
filter1 = filter1[(filter1['S1GRD0809'] != 3) & (filter1['S1GRD0809'] != 4) & (filter1['S1GRD0809'] > 0)]
len(filter1)

20242

## YES
This took a lot of trial and error to nail down, but finally the figure is correct!!

**From paper:**

"Our analytic sample is restricted to first-time 9th-graders, whose information on their career aspirations is available in at least one of the two survey rounds (n = 20,242)."

## Now going into STEM findings
Relevant variable here is: X1STU30OCC_STEM1: X1 Student occupation at age 30: STEM code 1 (sub-domain) (I believe this is from 9th grade?)

0	Not a STEM occupation	13,969	59.43

1	Life and Physical Science, Engineering, Mathematics, and Information Technology Occupations	1,726	7.34

4	Health Occupations	4,351	18.51

5	Split across 2 sub-domains	413	1.76

6	Unspecified sub-domain	301	1.28

9	Uncodeable	258	1.10

-9	Missing	426	1.81

-8	Unit non-response	2,059	8.76

As there is also:

X2STU30OCC_STEM1: X2 Student occupation at age 30: STEM code 1 (sub-domain)

0	Not a STEM occupation	12,991	55.27

1	Life and Physical Science, Engineering, Mathematics, and Information Technology Occupations	1,878	7.99

4	Health Occupations	4,619	19.65

5	Split across 2 sub-domains	601	2.56

6	Unspecified sub-domain	55	0.23

9	Uncodeable	96	0.41

-9	Missing	354	1.51

-8	Unit non-response	2,909	12.38

## Let's check Table B1
This is the product of the linear model they ran. Simple n counts to start! 

In [199]:
stem = filter1['X1STU30OCC_STEM1'].value_counts().reset_index()
s = sum(stem[stem['index'].isin([0,1,2,3,4,5,6])]['X1STU30OCC_STEM1'])
s

19633

Correct! Table B1 reports "n=19633" for 9th grade first round

In [200]:
stem = filter1['X2STU30OCC_STEM1'].value_counts().reset_index()
s = sum(stem[stem['index'].isin([0,1,2,3,4,5,6])]['X2STU30OCC_STEM1'])
s

17421

### Checking some percentages
Let's check that "Among first-time 9th graders in fall 2009, only about 11.4% of students were interested in pursuing a STEM career upon entering high school (see Figure 1)"

Shoot. We need to consider the weights!! W1STUDENT and W2W1STU

In [195]:
filter1['W2W1STU']

0        363.091114
1        193.466338
2        141.015287
3          0.000000
4        135.640155
            ...    
23498    146.107952
23499     13.462032
23500     84.709511
23501    197.490445
23502    143.927612
Name: W2W1STU, Length: 20242, dtype: float64

In [192]:
stem = filter1['X1STU30OCC_STEM1'].value_counts().reset_index()
s = sum(stem[stem['index'].isin([1,5])]['X1STU30OCC_STEM1'])
total = sum(stem[stem['index'].isin([0,1,2,3,4,5,6,9])]['X1STU30OCC_STEM1'])
s/total

0.1038666800926392

You will need it for your career